docs/learn-glossary.md

```diff
@@ -3,160 +3,175 @@ id: learn-glossary
-failures or timeouts and act accordingly. There is no limit on potential workflow
-duration.
+A fault-oblivious stateful function that orchestrates activities.
+
+- A Workflow has full control over which [Activities](#activity) are executed, and in which order.
+- A Workflow must not affect the external world directly, only through [Activities]((#activity)).
+- What makes Workflow code a Workflow is that its state is preserved by Temporal. Therefore any failure of a [Worker](#worker) process that hosts the Workflow code does not affect the [Workflow Execution](#workflow-execution). The Workflow continues as if these failures did not happen. At the same time, [Activities](#activity) can fail any moment for any reason.
+- Because Workflow code is fully fault-oblivious, it is guaranteed to get notifications about [Activity](#activity) failures or timeouts and act accordingly.
+- There is no limit to the duration of a Workflow.
 
 ### Workflow Execution
 
-An instance of a _Workflow_. The instance can be in the process of executing
-or it could have already completed execution.
+An instance of a [Workflow](#workflow).
+
+- The instance can be in the process of executing or it could have already completed execution.
 
-### Workflow Id
+### Workflow ID
 
-A unique identifier for a _Workflow Execution_. Temporal guarantees the
-uniqueness of an Id within a namespace. An attempt to start a _Workflow_ with a
-duplicate Id results in an **already started** error.
+A unique identifier for a [Workflow Execution](#workflow-execution).
+
+- Temporal guarantees the uniqueness of an ID within a [Namespace](#namespace).
+- An attempt to start a [Workflow](#workflow) with a duplicate ID results in an **already started** error.
 
 ### Workflow Task
 
-Synonym of the [Decision Task](#decision-task).
+A Workflow Task is a [Task](#task) that contains invocation information for a [Workfow](#workflow).
+
+- Every time a new external event that might affect a [Workflow](#workflow) state is recorded, a Workflow Task that contains it, is added to a [Workflow Task Queue](#workflow-task-queue) and then picked up by a [Workflow Worker](#worflow-worker).
+- After the new event is handled, the Workflow Task is completed with a list of [Commands](#command).
+- Handling of a Workflow Task is usually very fast and is not related to the duration of operations that the [Workflow](#workflow) invokes.
+
+### Workflow Task Queue
+
+[Task Queue](#task-queue) that is used to deliver [Workflow Tasks](#workflow-task) to [Workflow Workers](#workflow-worker)
 
 ### Workflow Worker
 
-An object that is executed in the client application and receives [decision tasks](#decision-task) from an [decision task queue](#decision-task-queue) it is subscribed to. Once task is received it is handled by a correponding workflow.
+A [Worker](#worker) service that receives [Workflow Tasks](#workflow-task) from an [Workflow Task Queue](#workflow-task-queue) it is subscribed to.
```

Comment:
In the SDK it is not exposed directly being a part of a Worker.

Reply:
Can you clarify your comment on this? I am not sure I totally understand what you mean -

Reply:
Per our conversation - consolidating into a single "Worker" glossary definition and removing "Activity Worker" and "Workflow Worker" terms.